lib/min_heap.rb

```diff
@@ -14,18 +14,24 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity:  generally O(log n) because of heap_up;
+  #                   O(n log n) if you have to make a new bigger store and move everything over
+  # Space Complexity: O(log n) because of heap_up
   def add(key, value = key)
```

Comment:
The time complexity is O(n log n) because you add n items to the heap and each add takes O(log n) time.